### ISSUE TITLE
[False positive] no-async-without-await and abstract methods

### ISSUE THREAD

**Issue by feimosi** [palantir/tslint#4773]
### Bug Report

- __TSLint version__: tslint@5.18.0
- __TypeScript version__: typescript@3.5.2
- __Running TSLint via__: CLI and VSCode

#### Reproduction using [TSLint Playground](https://palantir.github.io/tslint-playground)

I couldn't reproduce it in the Playground, however even invalid code is not detected by this rule what apparently means the rule is not present there at all (?)
[TSLint Playground](https://palantir.github.io/tslint-playground/?saved=N4Igxg9gJgpiBcID0SAEBlALgQwE6dUwE8AHASwDsBzVSwgCxlUllQCMIAPQiVKmAgGcc+GFAB0AHQrTKmGLgBm2MEwBKMEhEFlMEXEQA8AFQB8qYNNTXCMYQAoAlPFTGA3NIC+HmRRictfFRsNmFcFQIwABtsQUEMCABbGHQFADcyVRNzSwprElwIeTB5KFRcGGwoCAooomDQzHCS8s1tXX0iFw0tHT0DbJ8rVAKimBKxBrCI4MEiCjBUZSjBGAAFdswyNJh7MigXCgBXRLYFZ1Q1wsSyVZNUAB9UY6io0yG8kaO2KMzZ+cWTSO602212jgsqCQACpUNC0J4vNIQAAacA1RRkKgIZBoACaECOzGweUS0DIinqmEYzAxWKO4S2NVQjAqqAA7rp6KgiITcKgIOyKFIKChUABZbAAayYggZTGp2AIvIZqAAVoJmbdUGlsL8oCjghQyhSeYTykc8nIINIxUVWbQ4sDBIamvUwIwwFLKDRqUxIBRNVEmIp9KgFIVcIIRdJctZJCB-PJjYIEy4ANrDeMgTCCX4UTDwPVRBMorOoBO5-OYAC0AcxVBrwhJUDwUFL5crebkdbpjewZFwbAobBrbFiMATwwAumXPgncEdg6mEBZOyAKBAa7EATXOdTCbXsOyB5g04RF5PPp4oWgqmV77oyDU9Rbl0ayn7B7SKA2GUrn18REKBATwgA)

#### TypeScript code being linted

```ts
export abstract class SomeService<T> {
  protected readonly abstract repository: Repository<T>;

  protected abstract async falsePositive(id: number): Promise<T | null>;

  public async truePositive() { /* */ }
}
```

with `tslint.json` configuration:

```json
{
  "extends": [
    "tslint:all",
    "tslint-config-standard",
    "tslint-config-airbnb-base"
  ],
  "rules": {
    "no-async-without-await": true,
	"...": "..."
  }
}

```

#### Actual behavior
Treats abstract methods as an error

#### Expected behavior
Should ignore abstract methods because they don't have a body

**Comment by danielrentz** [comment]
Same for overloaded functions/methods, as described in #4776
